performance-tools/performance-cluster/setup-cluster.sh

```diff
@@ -0,0 +1,174 @@
+
+
+# shellcheck disable=SC1090
+source "${TEST_INFRA_PERFORMANCE_TOOLS_CLUSTER_SCRIPTS}/library.sh"
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+
+    key="$1"
+
+    case ${key} in
+        --grade-cluster)
+            checkInputParameterValue "$2"
+            GRADE_CLUSTER="$2"
+            shift # past argument
+            shift # past value
+        ;;
+        *)    # unknown option
+            POSITIONAL+=("$1") # save it in an array for later
+            shift # past argument
+        ;;
+    esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+if [[ "$GRADE_CLUSTER" != "production" ]]; then
+    export KYMA_CLUSTER_LABEL_VALUE="${GRADE_CLUSTER}"
+    export INPUT_CLUSTER_NAME="${GRADE_CLUSTER}"
+fi
+shout "Cluster Grade ${KYMA_CLUSTER_LABEL_VALUE}"
+
+function checkPodTerminated() {
+    local retry=0
+    local runningPods=0
+    local succeededPods=0
+    local failedPods=0
+
+    while [ "${retry}" -lt "${KYMA_CLUSTER_TIMEOUT_SEC}" ]; do
+        # check status phase for pod
+        for podName in $(kubectl get pods -n "${KYMA_CLUSTER_NAMESPACE}" -o json | jq -sr '.[]|.items[].metadata.name')
+        do
+            runningPods=$((runningPods + 1))
+            phase=$(kubectl get pod "${podName}" -n "${KYMA_CLUSTER_NAMESPACE}" -o json | jq '.status.phase')
+            echo "Test pod '${podName}' has phase: ${phase}"
+
+            if [[ "${phase}" == *"Succeeded"* ]]
+            then
+                succeededPods=$((succeededPods + 1))
+            fi
+
+            if [[ "${phase}" == *"Failed"* ]]; then
+                failedPods=$((failedPods + 1))
+            fi
+        done
+
+        # exit permanently if one of cluster has failed status
+        if [ "${failedPods}" -gt 0 ]
+        then
+            echo "${failedPods} pod(s) has failed status"
+            return 1
+        fi
+
+        # exit from function if each pod has succeeded status
+        if [ "${runningPods}" == "${succeededPods}" ]
+        then
+            echo "All pods in ${KYMA_CLUSTER_NAMESPACE} namespace have succeeded phase"
+            return 0
+        fi
+
+        # reset all counters and rerun checking
+        delta=$((runningPods-succeededPods))
+        echo "${delta} pod(s) in ${KYMA_CLUSTER_NAMESPACE} namespace have not terminated phase. Retry checking."
+        runningPods=0
+        succeededPods=0
+        retry=$((retry + 1))
+        sleep 5
+    done
+
+    echo "The maximum number of attempts: ${retry} has been reached"
+    return 1
+}
+
+setupCluster() {
+    shout "Install helm chart kyma-cluster "
+    date
+
+    if [  -f "$(helm home)/ca.pem" ]; then
+        local HELM_ARGS="--tls"
+    fi
+
+    helm install "${KYMA_CLUSTER_PATH}" \
+        --name "${KYMA_CLUSTER_NAME}" \
+        --namespace "${KYMA_CLUSTER_NAMESPACE}" \
+        --timeout "${KYMA_CLUSTER_HELM_TIMEOUT_SEC}" \
+        --wait ${HELM_ARGS} \
+        --set labels.grade="${KYMA_CLUSTER_LABEL_VALUE}" \
```

Comment:
We may not need to set all these values. Lets use configmaps and secrets for almost all of them.